### PR TITLE
Update coupled tests to use P7 configuration, add standalone P7 test suite

### LINKED ISSUE #39: point to icepack consortium master
NOAA-EMC CICE currently points to an EMC fork of Icepack because the non-benchmark IC for the current coupled regression tests would abort at this section of ``icepack/columnphysics/icepack_therm_itd.F90``.

```
              write(warnstr,*) subname, &
                 'ITD Thermodynamics: hicen_init(n+1) <= hicen_init(n)'
              call icepack_warnings_add(warnstr)
              call icepack_warnings_setabort(.true.)
```

NOAA-EMC has maintained a version with ``icepack_warnings_setabort(.false.)`` to prevent the code aborting on the first timestep.  

Testing of the new IC as part of ufs-weather-model PR #[765](https://github.com/ufs-community/ufs-weather-model/pull/765) has shown this flag is no longer required to be set ``.false.`` to prevent failures on the initial timestep. 

The flag should be set to true, because that will indicate that this specific condition is causing the model to abort.

### LINKED ISSUE #698: C384 P7 coupled tests fail on cheyenne
## Description

This job fails to run at startup with an error: ``MPT: shepherd terminated: r5i4n4.ib0.cheyenne.ucar.edu - job aborting``

### To Reproduce:

Cheyenne.intel

## Additional context

Found during testing for PR #639. The test was turned off for cheyenne.intel until the issue can be resolved. 

Since the wave model cannot be compile in debug mode, an equivalent test was created for a non-wave bmark_p7b configuration. The test can be accessed using rt.test in this [branch](https://github.com/DeniseWorthen/ufs-weather-model/tree/test2)

When running in debug mode, the model fails with the following:

```
2:MPT:     header=header@entry=0x7ffdcfe45c10 "MPT ERROR: Rank 32(g:32) received signal SIGFPE(8).\n\tProcess ID: 62280, Host: r7i7n1, Program: /glade/scratch/worthen/FV3_RT/rt_13027/cpld_bmark_v16_p7b/fv3.exe\n\tMPT Version: HPE MPT 2.22  03/31/20 15"...) at sig.c:340
32:MPT: #3  0x00002b177c55e4ff in first_arriver_handler (signo=signo@entry=8,
32:MPT:     stack_trace_sem=stack_trace_sem@entry=0x2b1786c00080) at sig.c:489
32:MPT: #4  0x00002b177c55e793 in slave_sig_handler (signo=8, siginfo=<optimized out>,
32:MPT:     extra=<optimized out>) at sig.c:565
32:MPT: #5  <signal handler called>
32:MPT: #6  0x0000000008047c67 in module_sf_noahmplsm::energy (parameters=..., ice=0,
32:MPT:     vegtyp=17, ist=1, nsnow=3, nsoil=4, isnow=0, dt=300,
32:MPT:     rhoair=1.1787577636261917, sfcprs=101003.09822185335,
32:MPT:     qair=0.011852218429786931, sfctmp=296.38129134030879,
32:MPT:     thair=296.38129134030879, lwdn=345.52102043841046, uu=3.4295500034750659,
32:MPT:     vv=-3.4672281739002062, zref=10.812501434103201,
32:MPT:     co2air=39.896223797632075, o2air=21109.64752836735, solad=..., solai=...,
32:MPT:     cosz=-0.91289500682162039, igs=1, eair=1910.8519305175637,
32:MPT:     tbot=299.26776123046875, zsnso=..., zsoil=..., elai=0, esai=0, fwet=0,
32:MPT:     foln=1, fveg=0, pahv=0, pahg=0, pahb=0, qsnow=0, dzsnso=...,
32:MPT:     lat=0.21661703129464016, canliq=0, canice=0, iloc=6, jloc=-9999,
32:MPT:     z0wrf=nan(0x7baddadbaddad), imelt=..., snicev=..., snliqv=..., epore=...,
32:MPT:     t2m=nan(0x7baddadbaddad), fsno=0, sav=0, sag=0,
32:MPT:     qmelt=nan(0x7baddadbaddad), fsa=0, fsr=0, taux=nan(0x7baddadbaddad),
32:MPT:     tauy=nan(0x7baddadbaddad), fira=nan(0x7baddadbaddad),
32:MPT:     fsh=nan(0x7baddadbaddad), fcev=nan(0x7baddadbaddad),
32:MPT:     fgev=nan(0x7baddadbaddad), fctr=nan(0x7baddadbaddad),
32:MPT:     trad=nan(0x7baddadbaddad), psn=nan(0x7baddadbaddad),
32:MPT:     apar=nan(0x7baddadbaddad), ssoil=nan(0x7baddadbaddad), btrani=...,
32:MPT:     btran=9.9999999999999995e-07, ponding=nan(0x7baddadbaddad),
32:MPT:     ts=nan(0x7baddadbaddad), latheav=nan(0x7baddadbaddad),
32:MPT:     latheag=nan(0x7baddadbaddad), frozen_canopy=3435973836,
32:MPT:     frozen_ground=3435973836, tv=291.71072387695312, tg=291.71072387695312,
32:MPT:     stc=..., snowh=0, eah=2000, tah=291.71072387695312, sneqvo=0, sneqv=0,
32:MPT:     sh2o=..., smc=..., snice=..., snliq=..., albold=0.65000000000000002, cm=0,
32:MPT:     ch=0, dx=-9999, dz8w=-9999, q2=0.011852218429786931, tauss=0, laisun=0,
32:MPT:     laisha=0, rb=0, errmsg=..., errflg=0, qc=-9999, qsfc=9.99e+20,
32:MPT:     psfc=101128.33749723693, t2mv=0, t2mb=nan(0x7baddadbaddad), fsrv=0,
32:MPT:     fsrg=0, rssun=nan(0x7baddadbaddad), rssha=nan(0x7baddadbaddad), albd=...,
32:MPT:     albi=..., albsnd=..., albsni=..., bgap=0, wgap=0,
32:MPT:     tgv=nan(0x7baddadbaddad), tgb=nan(0x7baddadbaddad),
32:MPT:     q1=nan(0x7baddadbaddad), q2v=0, q2b=nan(0x7baddadbaddad),
32:MPT:     q2e=nan(0x7baddadbaddad), chv=0, chb=nan(0x7baddadbaddad),
32:MPT:     emissi=nan(0x7baddadbaddad), pah=0, shg=0, shc=0,
32:MPT:     shb=nan(0x7baddadbaddad), evg=0, evb=nan(0x7baddadbaddad), ghv=0,
32:MPT:     ghb=nan(0x7baddadbaddad), irg=0, irc=0, irb=nan(0x7baddadbaddad), tr=0,
32:MPT:     evc=0, chleaf=0, chuc=0, chv2=0, chb2=nan(0x7baddadbaddad),
32:MPT:     .tmp.ERRMSG.len_V$698=512)
```

### LINKED ISSUE #763: 1deg CICE fix files are inconsistent with mesh and MOM6 runtime mask
## Description

The 1deg CICE fix files (``grid_cice_NEMS_mx100.nc`` and ``kmtu_cice_NEMS_mx100.nc``) do not match the mask which MOM6 creates at run time (see related MOM6 issue [47](https://github.com/NOAA-EMC/MOM6/issues/47)). This results in the history files produced by CICE 1deg  being incorrectly masked at the affected point.

The 1deg mesh file currently used is correct. 

### To Reproduce:

Examine the land mask variable ``kmtu`` in either of the fix files or examine a field in the 1deg history file.

This will be fixed with the updated input directory required for the new P7 coupled regression tests.

### LINKED ISSUE #766: Update coupled tests to use P7 configuration
## Description

The current low resolution coupled RTs are using GFS 2017 physics (v14). These tests need to be updated to match the configurations being tested in the coupled Prototypes.

## Solution

- Implement test configurations (c48mx400, c96mx100, c192mx050, c384mx025) matching P7 configuration.

- Add tests for threading, decomposition, debug and restart reproducibility for the the new C96mx100 configuration.

- Retain single "bmark" test which includes waves at C384mx025.

- Update non-benchmark RT test IC dates to match standalone global RTs (2021032206).

- Remove unused tests and configuration files.

### LINKED ISSUE #817: Add control_p7 test
## Description

We need a test for the standalone ATM which combines physics features planned for the coupled prototypes. This will allow the features to be tested for decomp, threading, restart, debug and reproducibility prior to being implemented in the coupled model.

## Solution

Add a test for ``APP=ATM`` using the P7 configuration.

## Related to

Issue #766

### PR DESCRIPTION
**Pull request by DeniseWorthen**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

- Updates coupled regression tests to use P7 configuration.  Retains a single benchmark configuration test ``cpld_bmark_p7`` for C384mx025 + waves using benchmark ICs (2013-04-01). Replaces all other coupled tests with the P7 configuration using 2021-03-22-06 ICs without waves.

- Adds a suite of P7 tests for the standalone ``app=ATM``. 
```
control_p7
control_restart_p7
control_debug_p7
control_decomp_p7
control_2threads_p7
```
The standalone and coupled P7 tests share the namelist ``cpld_control.nml.IN``.

- Uses a single compile (``app=S2SW``) for all coupled tests (except debug) and controls non-wave tests using namelist variables. 

- Removes ccpp suite files which are now unused (``FV3_GFS_2017_coupled``, ``suite_FV3_GFS_2017_couplednsst.xml`` and ``suite_FV3_GFS_2017_satmedmf_coupled.xml``). 

- Uses a new input data directory ``input-data-20210930``. Adds new ICs for MOM6 and CICE6 for ``2021-03-22-06``. Adds  Frac grid consistent ICs for the coupled model from ``/scratch2/NCEPDEV/ensemble/Bing.Fu/p7rt/merge`` on 20210825. These are placed into ``INPUT_L127_mxRES`` directories within the correct FV3_input_data directory where RES=ocean/ice resolution (400,100,050,025). These were updated with files from ``/scratch2/NCEPDEV/ensemble/Bing.Fu/p7rt/merge_fix/CRES_mxRES/2021032206/gfs/CRES/INPUT/`` with the ceil/floor bug fix on 09/13

- Changes  MOM6 setting ``MOM6_REPRO_LA`` to ``MOM6_USE_LI2016``. These are equivalent, but the new setting is more descriptive of the setting.

- Adds ice_in setting ``sec_init`` to accommodate initial time of 6 hours. 

- Sets ``FNSMCC   = global_soilmgldas.statsgo.t1534.3072.1536.grb`` for all configurations

- Removes NEMSDatm tests; associated settings and files in ``default_vars`` and ``parm`` will be removed in a follow-up PR.

- Removes standalone feature tests for features which have been incorporated into the P7 configuration (e.g. noahmp, merra2, ugwpv1).

- For pre-commit testing, the new coupled and standalone tests can be tested with ``rt.p7.conf`` using ``rt.sh -l rt.p7.conf ....`` on platforms which have the inputdata directory available.

<p>&nbsp;</p>

New input-data-directory (``input-data-20210930``) transfers:
- [x] hera
- [x] orion
- [x] cheyenne
- [x] gaea
- [x] jet
- [x] wcoss_cray
- [x] wcoss_dell_p3

### Issue(s) addressed

- fixes #766 
- fixes #763 
- fixes #817
- fixes https://github.com/NOAA-EMC/CICE/issues/39

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms).

- The input.nml created for the ``cpld_control_c384_p7`` test was compared to the input.nml for the existing ``cpld_bmark_wave_v16_p7b`` test. The ``fv_core_nml``, ``gfdl_cloud_microphysics_nml``,  and ``cires_ugwp_nml`` are functionally equivalent (e.g.  .T. vs .true.)

- The ``gfs_nml`` differences are shown graphically here:
cpld_bmark_wave_v16_p7b_vs_cpld_control_c384_p7

- All the non-global coupled RTs (DATM, HAFS, ATM, ATMW, ATMAERO) were run using the ``input-data-20210930`` at commit 19fa289e. All tests pass against the ``develop-20210915`` baseline with the exception of the DATM configurations. These do not pass because of the fix to the 1deg CICE grid file (#763) and updated CICE grid files created for the mx025 resolution. The currently used files were created on Theia on 20180927. The new files have been created using the updated gen_grid routine which will be added to UFS-Utils. The values of ulat,ulon changed because the conversion factor ``rad2deg`` now uses a specified constant for ``pi`` rather than ``pi= 4.0*atan(1.0)``.

- Testing against the ``develop-20210929`` baseline at commit 3d2b0a6 shows the following tests fail

Hera.gnu

```
Missing baselines:
cpld_control_p7 
cpld_debug_p7
control_p7
control_debug_p7
```

Hera.intel
```
all datm_cdeps tests (due to change in CICE grid file and/or fix for land mask at 1deg noted above)
all P7 tests (missing baseline)
```

<p>&nbsp;</p>

## Baseline Testing 

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] CI ; not run, changes in this PR need to be propagated to CI tests

## Dependencies
- Requires fixes for these issues:
1. Issue #770
2. Issue #743 
3. Issue #742 
4. Issue #774

- [x] FV3 PR [#819](https://github.com/ufs-community/ufs-weather-model/pull/819) 
- [x] FV3 PR [#397](https://github.com/NOAA-EMC/fv3atm/pull/397)
- [x] FV3 PR [#402](https://github.com/NOAA-EMC/fv3atm/pull/402) 
- [x] FV3 PR [#396](https://github.com/NOAA-EMC/fv3atm/pull/396)
- [x] Cice PR [#40](https://github.com/NOAA-EMC/CICE/pull/40)
- [x] FV3 PR [#404](https://github.com/NOAA-EMC/fv3atm/pull/404)

### CODE REVIEW

File: tests/fv3_conf/control_run.IN
Comment (MinsukJi-NOAA) [inline tests/fv3_conf/control_run.IN]: ${SUFFIX} cannot be deleted as it is used by the utest script (to be operational requirement test). For regression test, $SUFFIX is an empty string.

File: tests/tests/cpld_bmark_p7
Comment (MinsukJi-NOAA) [inline tests/tests/cpld_bmark_p7:60]: Isn't RESTART_N = FHMAX - FHROT? I think we need to be consistent throughout standalone, coupled, and etc.

File: tests/fv3_conf/control_run.IN
Comment (DeniseWorthen) [reply tests/fv3_conf/control_run.IN]: Will ${DEP_RUN}${SUFFIX} work?

Comment (MinsukJi-NOAA) [reply tests/fv3_conf/control_run.IN]: Yes, I think so.

File: tests/tests/cpld_bmark_p7
Comment (DeniseWorthen) [reply tests/tests/cpld_bmark_p7:60]: This test is actually the previous bmark (wave) test. So we're not restarting it, just writing a restart at the end of the run. Since FHMAX has changed, RESTART_N needs to be updated.

Comment (MinsukJi-NOAA) [reply tests/tests/cpld_bmark_p7:60]: Thanks for the explanation. In this case FHROT=0. So, I assume FHMAX-FHROT should not make a difference. I just thought being consistent may be beneficial because it can cause confusion down the line.

Comment (DeniseWorthen) [reply tests/tests/cpld_bmark_p7:60]: I see, I can make it FHMAX-FHROT.

**Comment by MinsukJi-NOAA** [comment]
Is the test control_p7 a replacement for control or an additional test?

**Comment by DeniseWorthen** [comment]
Right now, the control p7 tests are additional tests. Jun is going to remove feature tests from the control test list which are now all combined into the new control P7 tests.

File: tests/tests/cpld_restart_p7
Comment (MinsukJi-NOAA) [inline tests/tests/cpld_restart_p7:74]: Is there a reason why NSTF_NAME in cpld_restart_p7 is different from that defined in default_vars.sh (2,1,0,0,0) which is used by cpld_control_p7?
It seems that if cpld_restart_p7 uses (2,1,0,0,0), it does not reproduce cpld_control_p7.

Comment (DeniseWorthen) [reply tests/tests/cpld_restart_p7:74]: In restart, the setting is 2,0,0,0 because there should be no spin-up on a restart run.  See issue [695](https://github.com/ufs-community/ufs-weather-model/issues/695https://github.com/ufs-community/ufs-weather-model/issues/695).

Comment (junwang-noaa) [reply tests/tests/cpld_restart_p7:74]: I think since the new ICs from GFSv16 (2021032206) do have the nst fields, we don't need to use (2,1,0,0,0) to spin up nsst model for coldstart either, we can directly use (2,0,0,0,0) so this NSTF_NAME does not need to change for restart. But ICs for benchmark runs does not have the nst  fields, we do need the spin-up (2,1,0,0,0).

**Comment by MinsukJi-NOAA** [comment]
@DeniseWorthen Was restart of cpld_bmark_p7 tested?

File: tests/rt_gnu.p7.conf
Comment (climbfuji) [inline tests/rt_gnu.p7.conf]: Should this file be deleted before merging?

Comment (DeniseWorthen) [reply tests/rt_gnu.p7.conf]: yes, both of the p7.conf files will be deleted.

**Comment by DeniseWorthen** [comment]
@MinsukJi-NOAA The bmark test is now specifically the actual benchmark configuration, which includes waves. So we no longer have a restart test for the bmark test. 

There is a restart test for c384; this takes the place of the old bmark (c384) tests w/o waves. 

new ``control_c384_p7``+restart = old ``cpld_bmark_v16`` + restart (w/ updates to final P7 configuration)

new ``cpld_bmark_p7`` = old ``cpld_bmark_wave_v16_p7b`` (w/ updates to final P7 configuration)

**Comment by MinsukJi-NOAA** [comment]
Hmm. Okay. I got the restart reproducibility for `cpld_bmark_v16` in my PR 829. But I am not getting at the moment the restart reproducibility for `cpld_bmark_p7`. According to @DeniseWorthen's explanation, it seems that I should be testing `control_c384_p7` instead of `cpld_bmark_p7`.

@junwang-noaa, Should the operational requirement test be based off of `control_p7`, `cpld_bmark_p7`, or `control_c384_p7`? Hopefully, it will handle all of them in the future, but which one is of higher priority as of now?

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/718079436/20210930213016/ufs-weather-model
Please make changes and add the following label back:
hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/718079436/20210930214517/ufs-weather-model
Please make changes and add the following label back:
hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/718079436/20210930213229/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_29970
Baseline creation and move successful
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/718079436/20210930225518/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_20063
Test cpld_restart_c384_p7 009 failed failed 
Test cpld_restart_c384_p7 009 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-BL

**Review by climbfuji — COMMENTED** [review]
There are too many changes in this PR to go through every line. I don't see anything wrong and will approve once all the regression tests passed and the submodules are merged and updated correctly.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/718079436/20210930210010/ufs-weather-model
Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_305967
Baseline creation and move successful
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/718079436/20211001053330/ufs-weather-model
Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_350928
Test cpld_control_c192_p7 006 failed failed 
Test cpld_control_c192_p7 006 failed in run_test failed 
Please make changes and add the following label back:
orion-intel-BL

**Comment by MinsukJi-NOAA** [comment]
CI tests break because of the changes made to the regression test-related files in this PR. They will be fixed in the next PR via #829

**Review by junwang-noaa — APPROVED** [review]

**Review by MinsukJi-NOAA — APPROVED** [review]

**Comment by climbfuji** [comment]
fv3atm was merged, new hash is 799b157

**Review by climbfuji — APPROVED** [review]

**Pull request merged**
